### PR TITLE
Zcashd fails get tree

### PR DESCRIPTION
**Pull request by zancas**

### CODE REVIEW

**Comment by zancas** [comment]
This commit tightens the definition of "new" to mean, at _least_:

* has no commitment trees

File: zingolib/src/lightclient.rs [hunk 1]
```diff
@@ -1462,6 +1462,37 @@ impl LightClient {
...
 
+    async fn wallet_has_empty_commitment_trees(&self) -> bool {
+        self.wallet
+            .transaction_context
+            .transaction_metadata_set
+            .read()
+            .await
+            .witness_trees
+            .as_ref()
+            .is_some_and(|trees| {
```
Comment (fluidvanadium) [inline zingolib/src/lightclient.rs:1473]: why are these conditions seperate?

File: zingolib/src/lightclient.rs [hunk 2]
```diff
@@ -1462,6 +1462,37 @@ impl LightClient {
...
+                trees
+                    .witness_tree_orchard
+                    .max_leaf_position(0)
+                    .unwrap()
+                    .is_none()
+            })
```
Comment (fluidvanadium) [inline zingolib/src/lightclient.rs:1479]: could we have an && inside this brace?

**Comment by fluidvanadium** [comment]
superceded by #503

**Pull request closed without merging**